ISM43362/ISM43362.cpp

```diff
@@ -662,10 +662,13 @@ bool ISM43362::open(const char *type, int id, const char *addr, int port)
         rnglocalport = 49152;
     }
 
-    /* Set local port */
-    if (!(_parser.send("P2=%d", rnglocalport) && check_response())) {
-        debug_if(_ism_debug, "\tISM43362: open: P2 issue\n");
-        return false;
+    if (memcmp(type, "1", sizeof("1")) != 0) { /* TCP connection */
```

Comment:
Maybe could you do the comparison only once at the beginning of the function, and save the result to know if it is a TCP or UDP connection ? To avoid comparing many times ?

Reply:
It would be even better to change API to use `nsapi_protocol_t` like I already recommended in the original issue: https://github.com/ARMmbed/wifi-ism43362/issues/75#issuecomment-647301268